### ISSUE TITLE
Lint idea: argument of static method is always 'this'

### ISSUE THREAD

**Issue by ThomasdenH** [palantir/tslint#3189]
If a certain argument to a static method is always 'this', it can and probably should instead be replaced by an equivalent non-static method. This could be detected and potentially even fixed automatically.

An example:
```typescript
class Item {
    static printItem(i: Item) {
        console.log(i);
    }

    constructor() {
        Item.printItem(this);
    }
}
```

This could be replaced by:
```typescript
class Item {
    printItem() {
        console.log(this);
    }

    constructor() {
        this.printItem();
    }
}
```

**Comment by ajafff** [comment]
Some random thoughts:

* "argument is always 'this'"
  * what about `let self = this;`?
  * `Item.printItem(other)` could also be replaced with `other.printItem()`
  * if you need to avoid possibly overridden methods on the prototype chain you need something like `Item.prototype.printItem.call(other)` - very ugly IMO
* needs the type checker to find all call sites
* does it need to make sense of places where the function is passed as a reference?
  * `array.forEach(Item.printItem)`
  * `array.sort(Item.compare)`

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for new rules. See #4534. 😱

If you'd like to see this rule implemented, you have two choices:
* **Recommended:** Use [typescript-eslint](https://typescript-eslint.io) to lint your TypeScript code with ESLint ✅
* **Legacy:** Write it as a [custom TSLint rule](https://palantir.github.io/tslint/develop/custom-rules/) 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
